### PR TITLE
Updated setup_scm to use _dev if wanted

### LINKED ISSUE #1: The directory needs to be a git repo
I don't think we actually mention you need to have a git repo for setuptools_scm to work.

### LINKED ISSUE #11: Answering "Other" to license question raises exception
```
What open-source license would you like to use for your package?
1 - BSD 3-Clause
2 - GNU GPL v3+
3 - Apache Software License 2.0
4 - BSD 2-Clause
5 - Other
> license [1]: 5
Traceback (most recent call last):
  File "/Users/dstansby/miniconda3/bin/pieceofcake", line 8, in <module>
    sys.exit(main())
  File "/Users/dstansby/miniconda3/lib/python3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/Users/dstansby/miniconda3/lib/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/Users/dstansby/miniconda3/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/dstansby/miniconda3/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/Users/dstansby/miniconda3/lib/python3.8/site-packages/pieceofcake/main.py", line 96, in main
    values[key] = value[click.prompt(BOLD + '> ' + prompt + END,
IndexError: list index out of range
```

### PR DESCRIPTION
**Pull request by nabobalis**
Updated versioning to follow the bugfix in astropy.

Fixes #1 by adding text to the index page. I might also add a comment on the minimal page. 
Fixes #11 by updating the docs to use cookiecutter instead of pieceofcake which is the cause of this.

### CODE REVIEW

File: {{ cookiecutter.package_name }}/{{ cookiecutter.module_name }}/version.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        pass
+
+    return pieces
+
+
+major, minor, bugfix = split_version(version)
+
+del split_version  # clean up namespace.
+
+release = 'dev' not in version
```
Comment (Cadair) [inline {{ cookiecutter.package_name }}/{{ cookiecutter.module_name }}/version.py:40]: Think we can safely drop all this.
```suggestion
```

File: {{ cookiecutter.package_name }}/{{ cookiecutter.module_name }}/_dev/scm_version.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+import os.path
+
+try:
+    from setuptools_scm import get_version
+
+    version = get_version(root=os.path.join('..', '..'), relative_to=__file__)
+except ImportError:
+    raise ImportError('setuptools_scm not installed')
+except Exception as e:
+    raise ValueError(f'setuptools_scm broken with {e}')
```
Comment (Cadair) [inline {{ cookiecutter.package_name }}/{{ cookiecutter.module_name }}/_dev/scm_version.py:12]:
```suggestion
    raise
except Exception as e:
    raise ValueError(f'setuptools_scm can not determine version.') from e
```

**Comment by Cadair** [comment]
I wonder if this is a little to complex to add to this supposedly "minimal" template. On the other hand I think it's a really useful feature, and one which astropy and sunpy both want to use.

Maybe a middle ground would be have it opt-in with the cookiecutter.json. What do you think @astrofrog ?

**Comment by nabobalis** [comment]
Why the tests pass now? Not a clue.

File: {{ cookiecutter.package_name }}/MANIFEST.in [hunk 3]
```diff
@@ -15,3 +15,7 @@ prune docs/_build
 prune docs/api
 
 global-exclude *.pyc *.o
+
+# This subpackage is only used in development checkouts
+# and should not be included in built tarballs
+prune {{ cookiecutter.module_name }}/_dev
```
Comment (Cadair) [inline {{ cookiecutter.package_name }}/MANIFEST.in:21]:
```suggestion
{% if cookiecutter.setuptools_scm_git == 'y' %}

# This subpackage is only used in development checkouts
# and should not be included in built tarballs
prune {{ cookiecutter.module_name }}/_dev
{% endif %}
```

File: {{ cookiecutter.package_name }}/{{ cookiecutter.module_name }}/__init__.py [hunk 4]
```diff
@@ -1,6 +1,6 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
 
-from .version import __version__
+from .version import version as __version__
```
Comment (Cadair) [inline {{ cookiecutter.package_name }}/{{ cookiecutter.module_name }}/__init__.py:3]: If `scm_git` is false then we shouldn't have the `version.py` file? So what happens to this? does setuptools_scm generate this? if it doesn't can we still grab `__version__` from somewhere?

File: hooks/post_gen_project.py [hunk 5]
```diff
@@ -36,10 +35,19 @@ def process_license(license_name):
             os.remove(os.path.join(PROJECT_DIRECTORY, 'licenses', license_file))
 
 
+def process_version(setuptools_scm_git):
+    if setuptools_scm_git != "y":
+        remove_dir(os.path.join(PROJECT_DIRECTORY,
+                                '{{ cookiecutter.module_name }}',
+                                '_dev'))
```
Comment (Cadair) [inline hooks/post_gen_project.py:42]: I think you also need to remove `version.py` in the package root if this isn't being used?

Comment (nabobalis) [reply hooks/post_gen_project.py:42]: Yes.

File: {{ cookiecutter.package_name }}/{{ cookiecutter.module_name }}/__init__.py (hunk 4 above)
Comment (nabobalis) [reply {{ cookiecutter.package_name }}/{{ cookiecutter.module_name }}/__init__.py:3]: Will fix.

File: cookiecutter.json [hunk 6]
```diff
@@ -4,20 +4,30 @@
...
+        "BSD 2-Clause",
+        "Other"
+    ],
     "project_url": "",
-    "minimum_python_version": ["3.8", "3.7", "3.6", "3.5"],
+    "minimum_python_version": [
+        "3.8",
+        "3.9",
+        "3.7",
+        "3.6"
```
Comment (nabobalis) [inline cookiecutter.json:19]: Is it time?

Comment (Cadair) [reply cookiecutter.json:19]: from here probs not, it's only an option.

**Review by Cadair — COMMENTED** [review]
We should add a section to the docs describing both the manifiest and the dev stuff. This repo is primarily a docs repo, so getting that right is important. I can take a swing at that if you don't want to?

Otherwise looks good.

**Comment by nabobalis** [comment]
> We should add a section to the docs describing both the manifiest and the dev stuff. This repo is primarily a docs repo, so getting that right is important. I can take a swing at that if you don't want to?
> 
> Otherwise looks good.

What else do you want?

File: docs/minimal.rst [hunk 7]
```diff
@@ -214,6 +220,13 @@ You can find out more about the syntax of this file in
 `Specifying the files to distribute <https://docs.python.org/3.8/distutils/sourcedist.html#specifying-the-files-to-distribute>`_
 in the Python documentation.
 
+If you do use ``setuptools_scm``, you have to be aware that **everything** that is git versioned will be included **by default**.
+You will have to add::
+
+    prune <folder or files>
+
+to prevent this and slim down your final built package.
```
Comment (nabobalis) [inline docs/minimal.rst:228]: What else should I add here?

File: .gitignore [hunk 8]
```diff
@@ -132,4 +132,11 @@ venv.bak/
...
-.pyre/
\ No newline at end of file
+.pyre/
+
+# Vscode plugin
+.history
+
+# Just in case you want to test out the defaults in the repo
+packagename
+None/
```
Comment (nabobalis) [inline .gitignore:142]: This is created locally for me.

**Comment by Cadair** [comment]
I do plan to get to this.

**Comment by Cadair** [comment]
Superseded by #27

**Pull request closed without merging**